ReadingListApp/ReadingListApp/Model Controller/BookController.swift

```diff
@@ -0,0 +1,101 @@
+        
+        do {
+            let booksData = try propertyListEncoder.encode(books)
+            //Call the write(to: URL) function on the data you encoded computed property. The url you pass in should be an unwrapped version of the readingListURL property.
+            try booksData.write(to: url)
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+    
+    func loadFromPersistentStore() {
+       // Inside of a do-try-catch- block, unwrap the readingListURL property.
+        guard let readingListUrl = readingListURL, FileManager.default.fileExists(atPath: readingListUrl.path) else { return }
+        do {
+            //Still inside of the block, use the Data(contentsOf: URL) initializer to get access to the property list form of the books. Assign this data you initialize to a constant.
+            let data = try Data(contentsOf: readingListUrl)
+            //Initialize a PropertyListDecoder and assign it to a constant.
+            let decoder = PropertyListDecoder()
+            //Create a constant called decodedBooks. Set its value by calling the decode method on the property list decoder you just created, and passing in the type it should be decoded as, and the data constant you made a couple steps ago. (Hint: the type parameter to this function should be [Book].self)
+            let decodedBooks = try decoder.decode([Book].self, from: data)
+            //Set the value of the books property in the BookController to the decodedBooks you just made.
+            books = decodedBooks
+           
+        } catch {
+            //In the catch block, you should make an error message that is descriptive of what happened
+            print(error.localizedDescription)
+        }
+    }
+    
+    func create(withBook title: String, reasonToRead: String) {
+        let book = Book(title: title, reasonToRead: reasonToRead)
+        books.append(book)
+        saveToPersistentStore()
+    }
+    
+    func delete(book: Book) {
+        guard let index = books.index(of: book) else {return}
+        books.remove(at: index)
+        saveToPersistentStore()
+    }
+    
+    func updateHasBeenRead(for book: Book) {
+        guard let index = books.index(of: book) else {return}
+        //turn it from a constant to a variable
+        var book = books[index]
+        book.hasBeenRead = !book.hasBeenRead
+        //swap it.
+        books.remove(at: index)
+        books.insert(book, at: index)
+        saveToPersistentStore()
+    }
+    
+    func updateBook(for book: Book, with title: String, reasonToRead: String) {
+        guard let index = books.index(of: book) else {return}
+        var book = books[index]
+        book.title = title
+        book.reasonToRead = reasonToRead
+        books.remove(at: index)
+        books.insert(book, at: index)
```

Comment:
You can also apply same approach for update function, if you want to keep your code concise. Get the index and change values based on that index:
`books[index].title = title`
`books[index].reasonToRead = reasonToRead`